src/main/com/github/soerxpso/portalfix/PortalFix.java

```diff
@@ -160,6 +164,40 @@ public void run() {
 		}
 	}
 	
+	class Countdown extends BukkitRunnable {
+		int timeLeft, timeBetweenMessages;
+		Player p;
+		Countdown child = null;
+		
+		//timeLeft and timeBetweenMessages should be in seconds
+		public Countdown(int timeLeft, int timeBetweenMessages, Player p) {
+			this.timeLeft = timeLeft;
+			this.timeBetweenMessages = timeBetweenMessages;
+			this.p = p;
+		}
+		
+		public void run() {
+			p.sendMessage(doColors(Config.getCountdownMessage().replace("%0", String.valueOf(timeLeft / 20))));
+			if(timeLeft - timeBetweenMessages > 0) {
+				child = new Countdown(timeLeft - timeBetweenMessages, timeBetweenMessages, p);
+				child.runTaskLater(PortalFix.getPlugin(), timeBetweenMessages);
+			}
+		}
+		
+		//Devoted is not responsible for the possible results of taking life advice from method names
+		public void killSelfAndChildren() {
+			if(child == null) {
+				this.cancel();
+			}else {
+				child.killSelfAndChildren();
+			}
+		}
+	}
```

Comment:
Alright, I'll fix it tonight

On Mon, Feb 22, 2016, 8:02 AM Daniel Boston <notifications@github.com>
wrote:

> In src/main/com/github/soerxpso/portalfix/PortalFix.java
> <https://github.com/DevotedMC/PortalFix/pull/2#discussion_r53645776>:
>
> > +			p.sendMessage(doColors(Config.getCountdownMessage().replace("%0", String.valueOf(timeLeft / 20))));
> > +			if(timeLeft - timeBetweenMessages > 0) {
> > +				child = new Countdown(timeLeft - timeBetweenMessages, timeBetweenMessages, p);
> > +				child.runTaskLater(PortalFix.getPlugin(), timeBetweenMessages);
> > +			}
> > +		}
> > +		
> > +		//Devoted is not responsible for the possible results of taking life advice from method names
> > +		public void killSelfAndChildren() {
> > +			if(child == null) {
> > +				this.cancel();
> > +			}else {
> > +				child.killSelfAndChildren();
> > +			}
> > +		}
> > +	}
>
> Using the runTaskTimer should let you avoid this method. Although it is
> clever :P
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/DevotedMC/PortalFix/pull/2/files#r53645776>.
>